### PR TITLE
[camel-k] SyndesisHeaderStrategy not loaded into the registry (1.6.x)

### PR DESCRIPTION
**Pull request by lburgazzoli**

### CODE REVIEW

File: app/connector/http/pom.xml [hunk 1]
```diff
@@ -125,12 +125,7 @@
       <scope>test</scope>
     </dependency>
     <dependency>
-      <groupId>io.syndesis.integration</groupId>
-      <artifactId>integration-runtime</artifactId>
-      <scope>test</scope>
-    </dependency>
-    <dependency>
-      <groupId>org.hibernate</groupId>
+      <groupId>org.hibernate.validator</groupId>
```
Comment (zregvart) [inline app/connector/http/pom.xml]: This is part of the Hibernate validator upgrade, this should not be on `1.6.x` (#5143)

File: app/server/controller/src/main/java/io/syndesis/server/controller/ControllersConfigurationProperties.java [hunk 2]
```diff
@@ -29,6 +35,9 @@
     private int integrationStateCheckInterval = 60;
     private boolean exposeVia3scale;
 
+    @NestedConfigurationProperty
+    private CamelK camelk = new CamelK();
```
Comment (zregvart) [inline app/server/controller/src/main/java/io/syndesis/server/controller/ControllersConfigurationProperties.java]: Are this bits intended in this PR? The title is about `SyndesisHeaderStrategy` and they add a bit more than that.

**Comment by heiko-braun** [comment]
Thanks for your feedback @zregvart. I have added the `wip` label, awaiting @lburgazzoli 's response

**Comment by heiko-braun** [comment]
@lburgazzoli Feel free to remove `wip`, once @zregvart's questions are clarified. Without that, I am reluctant to bring it into 1.6.x

File: app/connector/http/pom.xml (hunk 1 above)
Comment (lburgazzoli) [reply app/connector/http/pom.xml]: good catch

File: app/server/controller/src/main/java/io/syndesis/server/controller/ControllersConfigurationProperties.java (hunk 2 above)
Comment (lburgazzoli) [reply app/server/controller/src/main/java/io/syndesis/server/controller/ControllersConfigurationProperties.java]: yes it is, as customizers are loaded using a property we may need this to tweak what customizers are loaded at runtime, now the header strategy customizer is included by default but it was not in the first iteration

**Comment by lburgazzoli** [comment]
@zregvart @heiko-braun PR updated

**Pull request merged**